app/workers/search/remove_from_index_worker.rb

```diff
@@ -0,0 +1,13 @@
+module Search
+  class RemoveFromIndexWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :medium_priority, retry: 10
+
+    def perform(index, key)
+      return unless index && key
```

Comment:
Just as a point of discussion: we were talking about less jobs silently failing in a past discussion. Is there any legitimate scenario where either of these can be `nil`?

Reply:
No, I don't think there is a legitimate scenario where either can be `nil`. To be honest, the only reason I tossed this in real quick was that I saw a trend in other workers of returning when data was missing. I think part of that trend was to go back later and raise more helpful errors if/when that happens. I'm all for dropping this line 💯

Reply:
I'm currently trying to championing the raising errors/logging that something unexpected happened approach, so whenever I see a silent return in a worker expect to at least ask a question. 😉

Reply:
I agree with @citizen428 and think we can remove this check, the `return unless` pattern is mostly used to check if objects loaded from the DB are still there when the job is run. I noticed this check wasn't there in the delayed job

Reply:
Updated!